crates/types/src/schema/v2.rs

```diff
@@ -0,0 +1,755 @@
+// Copyright (c) 2023 - 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::identifiers;
+use crate::identifiers::DeploymentId;
+use crate::invocation::{InvocationTargetType, ServiceType};
+use crate::schema::deployment::{DeliveryOptions, DeploymentType};
+use crate::schema::invocation_target::{InputRules, OutputRules};
+use crate::schema::openapi::ServiceOpenAPI;
+use crate::time::MillisSinceEpoch;
+use arc_swap::ArcSwapOption;
+use restate_serde_util::MapAsVecItem;
+use serde_with::serde_as;
+use std::collections::HashMap;
+use std::ops::RangeInclusive;
+use std::sync::Arc;
+use std::time::Duration;
+
+#[serde_as]
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct Deployment {
+    pub id: DeploymentId,
+    pub ty: DeploymentType,
+    pub delivery_options: DeliveryOptions,
+    pub supported_protocol_versions: RangeInclusive<i32>,
+    /// Declared SDK during discovery
+    pub sdk_version: Option<String>,
+    pub created_at: MillisSinceEpoch,
+
+    #[serde_as(as = "restate_serde_util::MapAsVec")]
+    pub services: HashMap<String, ServiceRevision>,
+}
+
+impl MapAsVecItem for Deployment {
+    type Key = DeploymentId;
+
+    fn key(&self) -> Self::Key {
+        self.id
+    }
+}
+
+#[serde_as]
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct ServiceRevision {
```

Comment:
There seems to exist `identifiers::ServiceRevision`. Maybe pick another name to avoid confusion.